### ISSUE TITLE
Create release drafter GitHub Actions workflow

### ISSUE THREAD

**Issue by CosimoTiger** [CosimoTiger/plugin-utilities#6]
1. Some hypothetical code is being worked on in a branch or on a fork – a feature, fix, refactor etc.
2. It ends up in the `develop` branch and finally passes all tests
3. Once the `develop` branch is ready, a new pull request is created
4. Once the PR is merged into `main`, the latest commit is tagged (e.g. via git commands, then pushed) with a version tag
5. The version tag triggers a workflow which packages and tests the code
6. If the code passes, the workflow drafts a release with uploaded JAR assets for use on JitPack
